starknet-e2e-test/src/starknet_sovereign.rs

```diff
@@ -0,0 +1,147 @@
+    CoreContractInitData, CoreContractState, OperatorTrait, ProxyInitializeData, ProxySupportTrait,
+    StarknetMessagingTrait,
+};
+use starknet_core_contract_client::{LocalWalletSignerMiddleware, StarknetCoreContractClient};
+use starknet_ff::FieldElement;
+use zaun_sandbox::unsafe_proxy::deploy_starknet_sovereign_behind_unsafe_proxy;
+use zaun_sandbox::EthereumSandbox;
+
+pub struct StarknetSovereign {
+    _sandbox: EthereumSandbox,
+    client: StarknetSovereignContractClient,
+}
+
+impl StarknetSovereign {
+    pub fn address(&self) -> Address {
+        self.client.address()
+    }
+
+    pub fn client(&self) -> Arc<LocalWalletSignerMiddleware> {
+        self.client.client()
+    }
+
+    pub async fn deploy() -> Self {
+        // Try to attach to an already running sandbox (GitHub CI case)
+        // otherwise spawn new sandbox instance
+        let sandbox = if let Ok(endpoint) = std::env::var("ANVIL_ENDPOINT") {
+            EthereumSandbox::attach(Some(endpoint)).expect("Failed to attach to sandbox")
+        } else {
+            EthereumSandbox::spawn(None)
+        };
+
+        let client = deploy_starknet_sovereign_behind_unsafe_proxy(sandbox.client())
+            .await
+            .expect("Failed to deploy starknet contract");
+
+        Self { _sandbox: sandbox, client }
+    }
+
+    pub async fn create_settlement_conf(&self, data_path: PathBuf) -> PathBuf {
+        let settlement_conf = EthereumConfig {
+            http_provider: self.client.client().provider().url().to_string(),
+            core_contracts: hex_str_from_bytes::<20, true>(self.client.address().0),
+            chain_id: self.client.client().get_chainid().await.expect("Failed to get sandbox chain ID").as_u64(),
+            poll_interval_ms: Some(10u64), // Default is 7s, we need to speed things up
+            ..Default::default()
+        };
+
+        let conf_path = data_path.join("eth-config.json");
+        let conf_file = File::create(&conf_path).expect("Failed to open file for writing");
+        serde_json::to_writer(conf_file, &settlement_conf).expect("Failed to write settlement config");
+
+        conf_path
+    }
+
+    pub async fn initialize_with(&self, init_data: CoreContractInitData) {
+        let data = ProxyInitializeData::<0> { sub_contract_addresses: [], eic_address: Default::default(), init_data };
+
+        self.client.initialize_with(data).await.expect("Failed to initialize");
+
+        self.client.register_operator(self.client.client().address()).await.expect("Failed to register operator");
+    }
+
+    pub async fn initialize(&self, program_hash: StarkFelt, config_hash: StarkFelt) {
+        self.initialize_with(CoreContractInitData {
+            program_hash: convert_felt_to_u256(program_hash),
+            config_hash: convert_felt_to_u256(config_hash),
+            ..Default::default()
+        })
+        .await;
+    }
+
+    pub async fn initialize_for_goerli(&self, block_number: StarkFelt, state_root: StarkFelt) {
```

Comment:
Yeah a bit of docstring could help understand what "initialize" implies for this specific struct. Coz what is happening is not that simple